### ISSUE TITLE
otel/metric: doc describing differences between Gauge and other metric types

### ISSUE THREAD

**Issue by jmhodges** [open-telemetry/opentelemetry-go#4801]
### Problem Statement

It would be nice to write down what differs between a Gauge, an UpAndDownCounter, and a Counter near their definitions. The typing on their factory methods on [Meter](https://pkg.go.dev/go.opentelemetry.io/otel/metric#Meter) make it difficult to understand at a glance what each is for.

As an old hand, I can guess at which to use, but even I was hoping to a quick explainer to confirm what I believed about them.

### Proposed Solution

Perhaps the [Measurements section](https://pkg.go.dev/go.opentelemetry.io/otel/metric#hdr-Measurements) of otel/metics could get some more verbiage on when to use each one.

#### Alternatives

A link to a more descriptive doc could be added to [Meter](https://pkg.go.dev/go.opentelemetry.io/otel/metric#Meter) or the [Measurements section](https://pkg.go.dev/go.opentelemetry.io/otel/metric#hdr-Measurements) instead.

**Comment by pellared** [comment]
Can you check if the description under https://opentelemetry.io/docs/instrumentation/go/manual/#metrics is helpful?

**Comment by jmhodges** [comment]
Yeah, those are a good start! The "non-additive" is maybe not super clear to folks, but it's definitely the flavor of what I was looking for.

**Comment by pellared** [comment]
Do you have any proposal on how to improve it? Can we close the issue?

**Comment by alliasgher** [comment]
Opened #8225 for this.
